src/includes/dinit-fstream.h

```diff
@@ -0,0 +1,370 @@
+#ifndef DINIT_FSTREAM_H_INCLUDED
+#define DINIT_FSTREAM_H_INCLUDED 1
+
+#include <dinit-iostream.h>
+
+/*
+ * dinit-fstream is a special purpose library which provides a interface for automatic
+ * file descriptor handling (opening and closing). It's recommended to use this library
+ * instead of using pure iostream library and handling file descriptor yourself.
+ *
+ * This library provides two clases for output and input: ofstream and ifstream.
+ * See documention for each class in top of it.
+ *
+ * All of the functionality of this library is located in "dio" namespace.
+ */
+
+namespace dio {
+
+// fstream_base is class which maintains the file path and current state about
+// opened file and provides open() functions to opening a file path and will
+// return file descriptor number or -1 on failure cases in output/input streams.
+// It's supposed to be used in internal ofstream/ifstream functionality.
+class fstream_base
+{
+    protected:
+    const char *path; // File path
+    bool are_we_open = false;
+
+    public:
+    // open() functions
+    // Open file descriptor based on passed file path and return fd or -1
+    int open(const char *path, const int flags) noexcept;
+
+    int open(const char *path, const int flags, const mode_t mode) noexcept;
+
+    // Is current stream's file descriptor open?
+    bool is_open() noexcept;
+
+    // Close passed file descriptor and returns -1 on SUCCESS
+    int close(const int fd) noexcept;
+};
+
+/*
+ * ofstream
```

Comment:
Hmm, I used these names because they make sense not because they got used by standard library. I researched about filebuf and you right.

is Closing up stdout/in/err fds in the end of program good? I guess it is.

Reply:
Closing the standard fds is fine, it will happen by default if you have global objects to represent those streams (because the destructor should close the stream, and the destructor is called automatically).